tooling/nargo/src/ops/execute.rs

```diff
@@ -228,15 +258,45 @@ pub fn execute_program_with_profiling<
     foreign_call_executor: &mut E,
 ) -> Result<(WitnessStack<F>, ProfilingSamples), NargoError<F>> {
     let profiling_active = true;
-    execute_program_inner(
+    match execute_program_inner(
         program,
         initial_witness,
         blackbox_solver,
         foreign_call_executor,
         profiling_active,
-    )
+    ) {
+        Ok((witness_stack, profiling_samples, ..)) => Ok((witness_stack, profiling_samples)),
+        Err(err) => Err(err),
+    }
+}
+pub(crate) fn execute_program_with_brillig_fuzzing<
+    F: AcirField,
+    B: BlackBoxFunctionSolver<F>,
+    E: ForeignCallExecutor<F>,
+>(
+    program: &Program<F>,
+    initial_witness: WitnessMap<F>,
+    blackbox_solver: &B,
+    foreign_call_executor: &mut E,
+    brillig_branch_to_feature_map: Option<&BranchToFeatureMap>,
```

Comment:
Related to another comment about this, if this method enables fuzzing then the map should also be present (and in fact we only call this method with `Some`, never with `None`):

```suggestion
    brillig_branch_to_feature_map: &BranchToFeatureMap,
```